app/models/pet.rb

```diff
@@ -10,4 +12,16 @@ def shelter_name
   def self.adoptable
     where(adoptable: true)
   end
+
+  def self.search(name)
+    where("name ILIKE ?", "%#{name}%")
+  end
+
+  def approve
+    self.update(adoptable: false)
+  end
+
+  def application_approved?
+    applications.map{|app| app.status}.include?("Approved")
```

Comment:
Similar to a previous comment, the more Railsy/AR way to do this would be `applications.where(status: "Approved").empty?` or you would actually probably want .present? so that if it's false, that means none are approved